Wasapi Library

egecansen opened the pull request:
This commit improves the ServiceGenerator by implementing the builder pattern for more flexible construction, along with a dockerized Foodplanner backend and MongoDB instance to provide a consistent environment for running unit tests.

- Implemented the builder pattern to ServiceGenerator.
- Dockerized the Foodplanner backend to ensure consistent test execution.
- Integrated a Dockerized MongoDB instance to provide a reliable database connection for unit tests.
- Added unit tests to verify the functionalities.

egecansen committed:
Builder Pattern implemented

egecansen committed:
Wasapi library

egecansen committed:
Create ci.yml

egecansen committed:
Update Caller.java

egecansen committed:
update docker yaml

egecansen committed:
Update .gitignore

egecansen committed:
Workflow & Caller class update

egecansen committed:
new features

- logout feature test added
- workflow yamls updated

egecansen committed:
Unit test improvements

egecansen committed:
Port change

egecansen committed:
Unit Test Improvements

- Updated the delete user endpoint

egecansen committed:
Update AppTest.java

egecansen committed:
Update AppTest.java

Umutayb commented on src/main/java/FoodPlanner.java:
```diff
@@ -0,0 +1,93 @@
+import retrofit2.Call;
+import retrofit2.http.*;
+
+public class FoodPlanner extends ApiUtilities {
```
This should not be in the main directory, move it to test.

Umutayb commented on src/main/java/exceptions/JavaUtilitiesException.java:
```diff
@@ -0,0 +1,15 @@
+     * Constructs a JavaUtilitiesException with the specified error message.
+     * @param errorMessage The error message to be associated with this exception.
+     */
+    public JavaUtilitiesException(String errorMessage) {super(errorMessage);}
```
Rename exception, this isn't JavaUtilities anymore.

Umutayb commented on src/main/java/models/GetUserResponseModel.java:
```diff
@@ -0,0 +1,128 @@
+
+import java.util.List;
+
+public class GetUserResponseModel {
```
Move to test! Only library classes should be located in main, classes used for testing should always be placed in test directory.

Umutayb commented on src/main/java/models/SimpleMessageResponseModel.java:
```diff
@@ -0,0 +1,9 @@
+package models;
+
+public class SimpleMessageResponseModel {
```
Not sure what this is, supposed to be in test?

Umutayb commented on src/main/java/models/UserAuthRequestModel.java:
```diff
@@ -0,0 +1,27 @@
+package models;
+
+public class UserAuthRequestModel {
```
Wrong directory....

Umutayb commented on src/main/java/models/UserAuthResponseModel.java:
```diff
@@ -0,0 +1,33 @@
+
+import java.util.List;
+
+public class UserAuthResponseModel {
```
Wrong directory....

Umutayb commented on src/main/java/ApiUtilities.java:
```diff
@@ -0,0 +1,276 @@
+import static utils.reflection.ReflectionUtilities.iterativeConditionalInvocation;
+
+@SuppressWarnings("unused")
+public abstract class ApiUtilities extends Caller {
```
Please write a javadoc explanation of the class here. Also address Inna and myself as the original authors, and explain what changes with this version.

For examples check ContextStore javadocs.

Umutayb commented on src/main/java/ServiceGenerator.java:
```diff
@@ -0,0 +1,509 @@
+ * and configurations. It also provides methods to set and get different configurations like headers, timeouts,
+ * and base URL.
+ *
+ * @author Umut Ay Bora
```
Please update the javadoc explanation of the class here. Also explain what changes with this version (give yourself the credit for implementing the builder pattern).

For examples check ContextStore javadocs.

Umutayb commented on src/main/java/models/UserSignUpModel.java:
```diff
@@ -0,0 +1,58 @@
+
+import java.util.List;
+
+public class UserSignUpModel {
```
Wrong directory....

Umutayb commented on src/test/resources/application.properties:
```diff
@@ -0,0 +1,15 @@
+spring.application.name=food-planner
+spring.data.mongodb.authentication-database=admin
+spring.data.mongodb.database=food-planner
+spring.data.mongodb.uri=spring.data.mongodb.uri=mongodb://zan:Jbowmy0JFri54by@mongodb:27017/food-planner?authSource=admin
```
Update these credentials with generic ones (user, Secret-123).

Umutayb commented on docker-compose.yml:
```diff
@@ -0,0 +1,41 @@
+version: '3.8'
```
Update these credentials with generic ones (user, Secret-123).

Umutayb commented on mongo-init.js:
```diff
@@ -0,0 +1,71 @@
+db = db.getSiblingDB("food-planner");
```
Bury this file in the docker image.

Umutayb reviewed (changes requested):
Thank you for the refactor! Please read my change requests, some improvements are necessary before publishing.

Umutayb committed:
Project setup

Added CI Infrastructure.
Added a temporary readme, it needs refinement.
Moved classes to their appropriate directories.
Added JavaDocs to classes.

Umutayb committed:
Update pom.xml

Umutayb committed:
Refactoring ServiceGenerator & tests

Umutayb committed:
Builder Pattern

Wrapping up the initial release by:
Completing the refactor of ServiceGenerator (now WasapiClient).
Postponing unit tests to the subsequent release.

Umutayb committed:
Deleted these unit test files
